packages/nimble-components/src/table/specs/table-data-api.md

```diff
@@ -0,0 +1,130 @@
+
+```ts
+const rowTemplate = (index: number): ViewTemplate<any, Table<Person>> => html<
+    any,
+    Table<Person>
+>`
+    <nimble-table
+        style="max-height: 500px"
+        :data="${(_, c) => c.parent.tableData[index]!.row.original.friends}"
+    >
+    </nimble-table>
+`;
+```
+
+Both of these are demonstrated in the following prototype branch:
+
+[Table in a Table](https://github.com/ni/nimble/blob/tanstack-table-in-table) (see [Storybook](https://60e89457a987cf003efc0a5b-wzissaavew.chromatic.com/iframe.html?args=&id=table--table-story&viewMode=story))
+
+### Limitations
+
+Despite the generic typing of `Table` making it more strict, users would not benefit from that in a scenario where they could provide declarative column definitions that could theoretically only visualize a specific datatype. Take the following markup:
+
+template (Angular)
+
+```html
+<nimble-table [data]="data" #table>
+    <nimble-text-field-column columnId="firstName"></nimble-text-field-column>
+    <nimble-text-field-column columnId="lastName"></nimble-text-field-column>
+    <nimble-number-field-column columnId="age"></nimble-number-field-column>
+    <nimble-number-field-column
+        columnId="birthDate"
+    ></nimble-number-field-column>
+</nimble-table>
+```
+
+component (Angular)
+
+```ts
+@ViewChild('table') private table: Table<Person>;
+```
+
+Here, if the `birthDate` value was really a `DateTime`, but the provided column definition would try to render it as a `number`, the user would be unaware of the mistake by static type checking (in environments like Angular).
+
+### Angular Integration
+
+The Angular directive for the table will be generic for the row data type, `TData`. The `data` property will be exposed through the directive and have type of `TData[]`.
+
+### Blazor Integration
+
+The Blazor wrapper around the table will require writing interop code to set the property on the underlying `nimble-table` component because Blazor does not allow binding to properties on an element. This can either be done through a getter and setter for a `Data` parameter where the interop Task is not awaited, or it can be done by exposing an async method on the Blazor component that allows setting the data.
+
+### Hierarchical Data
+
+There are no special data API considerations for hierarchical data. All data will be specified as a flat array. Parent/child relationships will be specified using IDs (likely the child having a reference to its parent's ID). The details of hierarchical data are out of scope of this spec.
+
+### Data modification
+
+The API for modifying the table's data by manipulating the table is out of scope for this spec. An example of this is an editable numeric column where the user can modify the value associated with a row directly within the table.
+
+## Alternative Implementations / Designs
+
```

Comment:
Some more alternatives to consider mentioning, mostly lifted from the Table Architecture Spark:

1. the table won't initially support virtualized data. It would be up to clients to update the data array when the user reaches the end of the data set.
2. the table won't initially support updating a subset of the data. It will only detect a new array being assigned.